### ISSUE TITLE
Configure with monorepository 

### ISSUE THREAD

**Issue by Ahryman40k** [hbenl/vscode-mocha-test-adapter#114]
Hi, 

I have a lerna project with several tested typescript libraries.
But I'm unable to make tests discovered. 

In the default settings, no tests are found.
So I tried to set some parameters: 
```
    "mochaExplorer.files": "**/*.spec.ts", // or  "packages/**/*.spec.ts" or  "packages/<package>/src**/*.spec.ts"
    "mochaExplorer.logpanel": true,
    "mochaExplorer.pruneFiles": true,
    "mochaExplorer.require": [
        "ts-node/register", 
        "source-map-support/register"
    ]
```
I also tried to set "mochaExplorer.cwd" to a specific package root. ( packages/<package> )
But it act like my tsconfig wasn't use, meaning 

- it can't find tests dependencies when they come from other libraries inside my lerna project.
- it failed on chai/mocha/sinon imports with esModuleInterop error ...

Hope you may help me !
Ahry.

**Comment by hbenl** [comment]
Have you tried creating a [multi-root workspace](https://code.visualstudio.com/docs/editor/multi-root-workspaces) for your project? That usually works a lot better because VS Code will then understand that each package has its own settings (I mean `.vscode/settings.json` as well as `tsconfig.json` etc.).
Also, you probably should remove `source-map-support/register` from your requires because that is only needed if you run the tests from transpiled `*.spec.js` files instead of the original `*.spec.ts` files. But I don't think this is the source of your problems.

**Comment by liam-murray-xealth** [comment]
I recommend the following extension to make it easy to switch among packages in your mono repo. It makes mono repos work well with this extension.

https://marketplace.visualstudio.com/items?itemName=folke.vscode-monorepo-workspace

**Comment by xenoterracide** [comment]
also

```
    "mochaExplorer.ignore": ["**/node_modules/**","**/dist/**"],
    "mochaExplorer.watch": ["**/*.ts","package-lock.json"],
    "mochaExplorer.require": "ts-node/register",
    "mochaExplorer.files": "src/**/*.spec.ts",
 ```

**Comment by xenoterracide** [comment]
sorry about the comment spam, work proxy blocks *edit*
```
Darwin 5c52309d33e3 20.6.0 Darwin Kernel Version 20.6.0: Tue Feb 22 21:10:41 PST 2022; root:xnu-7195.141.26~1/RELEASE_X86_64 x86_64 i386 MacBookPro16,1 Darwin
1.67.2
c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
x64
ts-node@10.7.0 dev
```
test explorer
```
v2.13.5
```

**Comment by hbenl** [comment]
@xenoterracide I can see that the extension uses its bundled version of mocha instead of the version installed in your project, and this causes issues with `mochaExplorer.require` (which I want to fix, but I don't know yet how). Do you use a monorepo? If so, do you use a multi-root workspace as suggested above?

**Comment by xenoterracide** [comment]
yes, I set up the multi root workspace. I don't think that should be an issue.

**Comment by xenoterracide** [comment]
it only creates a version in your workspace if it needs to. Each workspace does not have it's own package-lock.json. this doesn't document that, but it's the workspace tool we're currently using. https://docs.npmjs.com/cli/v8/using-npm/workspaces yarn, IIRC wouldn't do that either, especially not if you're using pnp, which doesn't? drop them in node_modules at all.

**Comment by xenoterracide** [comment]
also might help to understand module resolution https://nodejs.org/api/modules.html#loading-from-node_modules-folders

**Comment by xenoterracide** [comment]
btw, sorry about the comment spam, work proxy doesn't allow us to edit comments

**Comment by xenoterracide** [comment]
also, I think I may actually have figured out how to fix this a long time ago, but don't ask me how. I'm sure I had to do something beyond the monorepo plugin though...

**Comment by hbenl** [comment]
Thanks, I hadn't really looked into npm/yarn workspaces before but now I understand why this extension keeps having problems with monorepos. I'll have a fix in the next couple of days.

**Comment by tjprescott** [comment]
I have the opposite problem. The mocha test explorer works great... but I can't get any of the tests to run with just `pnpm mocha`. I guess it's less crucial, but it's frustrating since I feel like both should work (they worked when there was just one package!)
